### ISSUE TITLE
[feat] Introduce occasional validated UTXO/prune blocks

### ISSUE THREAD

**Issue by bartdeboer** [dogecoin/dogecoin#2016]
Propose to introduce occasional prune/UTXO blocks into the blockchain. These would be large blocks that consolidate transaction history for certain sets of addresses. Such blocks could occur daily, weekly or monthly. To reduce stress they could rotate certain sets of addresses (or focus on the most active addresses). Such blocks could be used by nodes to update themselves more quickly into a verified pruned state without needing to download the entire transaction history. These blocks could be more easily listed on various (trusted) places (off chain) for easy downloading and ability to verify (the checksums) acting as sort of well-known security beacons in time.

Once the pruned state is reached the node can still start validating its pruned state by checking random addresses (or slowly checking the entire history). People would be able to store certain prune blocks they trust locally and use them to quickly spin up new nodes (on other devices) using different software clients. They could be shared between trusted parties.

**Comment by patricklodder** [comment]
Would this be better than the AssumeUTXO work being done for Bitcoin? See that work here: https://github.com/bitcoin/bitcoin/projects/11

**Comment by bartdeboer** [comment]
Yes, it's in a similar direction as having a trusted UTXO snapshots which are shared throughout the network. Though I would propose to add such snapshots to the blockchain (needing to be mined) making them part of the protocol (independent of software clients). As the blockchain grows larger we might reach a point in the future where people don't care to validate back into history for years and years all the way back to block 0. At some point the oldest blocks might just simply get "forgotten". The solution should allow for this I think. 

You could say Bitcoin and Dogecoin already have a theoretical UTXO snapshot at block 0. We're just adding extra ones.

**Comment by bartdeboer** [comment]
I mean the solution should allow for forgetting older blocks without compromising integrity. It would be up to the user how far back they'd care to validate the chain.

I'm willing to help wherever I can find the time. I'll have a look into AssumeUtxo to see if there's some way I can be of assistance for them.
